crest/Assets/Crest/Crest/Scripts/LodData/RegisterLodDataInput.cs

```diff
@@ -35,7 +36,7 @@ public interface ILodDataInput
     /// <summary>
     /// Base class for scripts that register input to the various LOD data types.
     /// </summary>
-    public abstract class RegisterLodDataInputBase : MonoBehaviour, ILodDataInput, IValidated
+    public abstract partial class RegisterLodDataInputBase : MonoBehaviour, ILodDataInput
```

Comment:
i'll defer to you guys. i like what this aims to achieve, but get your points. 

is there inspiration to take from *UnderwaterPostProcessHDRP.cs* and *UnderwaterPostProcessUtils.cs* - would pushing out some functions to a separate file work?

Reply:
Yeah, the reason I built those in that weird way is because of how different post-processing is in BIRP and SRP, whilst being able to have shared code inside a static utils class. It's not something that I feel super strongly about, just mainly stating why I would probably never make a class partial myself. I'm more than happy to work with them.

Reply:
> no longer are guarenteed to have everything related to a class in the same place and you start having cross-references and interdependencies

I would counter with that it helps separate concerns and allows us to exclude validation code from builds. Providing we use it responsibly it should be an improvement overall.

> Also, this won't compile in non-editor builds because "CheckShaderName" is now defined in an #if UNITY_EDITOR block, but is still being called from here

Fixed. It would be nice if we could try and keep editor specific code outside of standalone. The concern valid though. If we don't test by building to standalone it is possible to get caught out. But this would be an issue if we wrapped that method with `UNITY_EDITOR` too.

Reply:
The "modern" (not nessecarily better, but could be in this instance) way of doing this would be to put editor-only code inside a seperate asmdef. 

https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html

In this isntane it's probably the right thing to do as it prevents editor-code from accidentally being used in gameplay code.

Reply:
Good idea. I'll give that route a try.

Reply:
Partial classes must be within the same assembly. I could only do assemblies for the editors and I think that should be done all at once in a separate PR (there are other editors that could be part of that assembly).

The other issue is that we do call _Validate_ in _Awake_ for some of these components which wouldn't work with an assembly. It isn't mandatory but could be useful to keep that behaviour.

I also had a look at extension methods, but they cannot access private variables. To get assemblies to work we would need to at the very least have to start making more properties public (only getter). It wouldn't be such a bad thing though.

It is possible to do something like the following if you want them merged together:
```csharp
public partial class AssignLayer : MonoBehaviour
#if UNITY_EDITOR
    , IValidated
#endif
```

But this won't protect us from accidentally using editor code in build code.

Reply:
Unfortuantely asmrefs aren't a think in 2018.4 otherwise I would say to use that instead.

I will hav a look at the PR later this weekend :)

Reply:
I had a look at asmrefs but I couldn't see how they would help.